packages/spectral/src/serverTypes/integration.ts

```diff
@@ -12,13 +12,13 @@ export interface ComponentReference {
 
 export type Input =
   | {
-      name?: string;
+      name?: string | Input;
```

Comment:
This could allow an infinite nesting within `name` right? Should we hoist `name`'s type up so we can enforce only one level?

Reply:
It would. It's used as a guide for the conversion layer so I don't think avoiding the possible recursion is necessary (since it isn't on the DX "authoring" side).